src/components/selectable/selectable_list/selectable_list_item.tsx

```diff
@@ -18,12 +18,18 @@ import { EuiBadge, EuiBadgeProps } from '../../badge';
 function resolveIconAndColor(
   checked: EuiSelectableOptionCheckedType
 ): { icon: IconType; color?: IconColor } {
-  if (!checked) {
-    return { icon: 'empty' };
+  switch (checked) {
+    case undefined:
+      return { icon: 'empty' };
+    case 'on':
+      return { icon: 'check', color: 'text' };
+    case 'off':
+      return { icon: 'cross', color: 'text' };
+    case 'mixed':
+      return { icon: 'minus', color: 'text' };
+    default:
+      throw new Error('Icon must be enum "on" | "off" | "mixed" | undefined.');
   }
```

Comment:
Just my 2c, but rather than throwing, I think we should just gracefully degrade/default to unchecked:
```suggestion
  switch (checked) {
    case 'on':
      return { icon: 'check', color: 'text' };
    case 'off':
      return { icon: 'cross', color: 'text' };
    case 'mixed':
      return { icon: 'minus', color: 'text' };
    case undefined:
    default:
      return { icon: 'empty' };
  }
```